controllers/activemqartemis_reconciler.go

```diff
@@ -1229,6 +1232,33 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) deleteRequestedResource(customR
 	return deleteError
 }
 
+func checkExistingPersistentVolumes(instance *brokerv1beta1.ActiveMQArtemis, client rtclient.Client) {
+	var log = ctrl.Log.WithName("controller_v1beta1activemqartemis")
+
+	var i int32
+	for i = 0; i < instance.Spec.DeploymentPlan.Size; i++ {
+		ordinalString := strconv.Itoa(int(i))
+		pvcKey := types.NamespacedName{Namespace: instance.Namespace, Name: instance.Name + "-" + namer.CrToSS(instance.Name) + "-" + ordinalString}
+		pvc := &corev1.PersistentVolumeClaim{}
+		err := client.Get(context.TODO(), pvcKey, pvc)
+
+		if err == nil {
+			owners := pvc.OwnerReferences
+			if len(owners) > 0 {
+				log.Info("removing owner ref from pvc to avoid data loss")
```

Comment:
@gaohoward It's not for sure that this issue would cause data loss. It is only in the case that the PV has not been configured to 'Retain'. If it were configured to Retain then the PV would enter a state 'Released' when deleted where it was unavailable to be re-used, but the data would be sitting there awaiting manual human intervention.

So I think this statement is just a bit strong in the sense that we are removing the owner ref to avoid **potential** data loss, not guaranteed data loss. The difference is subtle but important I think.

Reply:
good point. +1